src/redux/actions/projects/saveProject.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-param-reassign */
+import fetchAPI from '../../../utils/fetchAPI';
+import pushNotificationMessage from '../notifications';
+import messages from '../../../components/notification/messages';
+import {
+  PROJECTS_ERROR,
+  PROJECTS_SAVING,
+  PROJECTS_SAVED,
+} from '../../actionTypes/projects';
+
+import errorTypes from './errorTypes';
+
+const saveProject = (projectUuid, newProject) => async (dispatch, getState) => {
+  const project = newProject ?? getState().projects[projectUuid];
+
+  dispatch({
+    type: PROJECTS_SAVING,
+  });
```

Comment:
make sure that after we dispatch this, the relevant data management component enters loading state